src/client/state/navigation.js

```diff
@@ -59,23 +59,10 @@ class Navigation extends EventEmitter {
       return;
     }
 
-    const parents = roomList.roomIdToParents.get(roomId);
-    if (!parents) return;
-    if (parents.has(this.selectedSpaceId)) {
-      this.spaceToRoom.set(this.selectedSpaceId, {
-        roomId,
-        timestamp: Date.now(),
-      });
-    } else if (accountData.categorizedSpaces.has(this.selectedSpaceId)) {
-      const categories = roomList.getCategorizedSpaces([this.selectedSpaceId]);
-      const parent = [...parents].find((pId) => categories.has(pId));
-      if (parent) {
-        this.spaceToRoom.set(parent, {
-          roomId,
-          timestamp: Date.now(),
-        });
-      }
-    }
```

Comment:
What this portion does is it check parent parent for room in when current selected space is mark as categorized. I looks to find the possible parent in those spaces(aka spaces that are displayed as categories). 

I don't know why we want to remove this behaviour as it will break it.

Reply:
The issue with this logic is that when there's a shared space nested in 2 separate spaces selecting a room from the shared space in one will mark it as last selected in the other one as well. I find this behavior really undesirable.

Consider the following layout:
Space `A` containing room `a` and space `S`
Space `B` containing room `b` and space `S`
Space `S` containing room `s`

Let's assume that spaces `A` and `B` are pinned to the sidebar with caregorized spaces enabled. You click on space `A` and select room `a`, then you open space `B` and select nested room `s`. Finally you go back to space `A` instead of expected room `a` you see room `s`.

I know my patch is really half-assed, but I have been running with it for several months with no issues so far.